crates/next-api/src/entrypoints.rs

```diff
@@ -6,7 +6,7 @@ use crate::{
     route::{Endpoint, Route},
 };
 
-#[turbo_tasks::value(shared)]
+#[turbo_tasks::value(shared, local)]
 pub struct Entrypoints {
     pub routes: FxIndexMap<RcStr, Route>,
     pub middleware: Option<Middleware>,
```

Comment:
Yeah, these aren't always obvious because they can be the result of transitively included `Vc`s.

- The `Route` type contains `AppPageRoute`, which contains two Vc fields: https://github.com/vercel/next.js/blob/92979e03c67d98aaea468388ebf1c59d82468dbb/crates/next-api/src/route.rs#L14-L15

- `Middleware` contains a `Vc` field: https://github.com/vercel/next.js/blob/92979e03c67d98aaea468388ebf1c59d82468dbb/crates/next-api/src/project.rs#L206

- `Instrumentation` contains a couple `Vc` fields: https://github.com/vercel/next.js/blob/92979e03c67d98aaea468388ebf1c59d82468dbb/crates/next-api/src/project.rs#L211-L212

Here's the error messages you get if you remove the `local`:

```
error[E0277]: the trait bound `route::Route: NonLocalValue` is not satisfied
  --> crates/next-api/src/entrypoints.rs:11:17
   |
11 |     pub routes: FxIndexMap<RcStr, Route>,
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^ the trait `NonLocalValue` is not implemented for `route::Route`, which is required by `IndexMap<turbo_rcstr::RcStr, route::Route, BuildHasherDefault<rustc_hash::FxHasher>>: NonLocalValue`
   |
   = help: the following other types implement trait `NonLocalValue`:
             &T
             &mut T
             ()
             (A, Z, Y, X, W, V, U, T)
             (B, A, Z, Y, X, W, V, U, T)
             (C, B, A, Z, Y, X, W, V, U, T)
             (D, C, B, A, Z, Y, X, W, V, U, T)
             (E, D, C, B, A, Z, Y, X, W, V, U, T)
           and 665 others
   = note: required for `IndexMap<turbo_rcstr::RcStr, route::Route, BuildHasherDefault<rustc_hash::FxHasher>>` to implement `NonLocalValue`
note: required by a bound in `entrypoints::_::{closure#0}::DeriveNonLocalValueAssertion::assert_impl_NonLocalValue`
  --> crates/next-api/src/entrypoints.rs:9:1
   |
9  | #[turbo_tasks::value(shared)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `DeriveNonLocalValueAssertion::assert_impl_NonLocalValue`
   = note: this error originates in the derive macro `turbo_tasks::NonLocalValue` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `project::Middleware: NonLocalValue` is not satisfied
  --> crates/next-api/src/entrypoints.rs:12:21
   |
12 |     pub middleware: Option<Middleware>,
   |                     ^^^^^^^^^^^^^^^^^^ the trait `NonLocalValue` is not implemented for `project::Middleware`, which is required by `std::option::Option<project::Middleware>: NonLocalValue`
   |
   = help: the following other types implement trait `NonLocalValue`:
             &T
             &mut T
             ()
             (A, Z, Y, X, W, V, U, T)
             (B, A, Z, Y, X, W, V, U, T)
             (C, B, A, Z, Y, X, W, V, U, T)
             (D, C, B, A, Z, Y, X, W, V, U, T)
             (E, D, C, B, A, Z, Y, X, W, V, U, T)
           and 665 others
   = note: required for `std::option::Option<project::Middleware>` to implement `NonLocalValue`
note: required by a bound in `entrypoints::_::{closure#0}::DeriveNonLocalValueAssertion::assert_impl_NonLocalValue`
  --> crates/next-api/src/entrypoints.rs:9:1
   |
9  | #[turbo_tasks::value(shared)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `DeriveNonLocalValueAssertion::assert_impl_NonLocalValue`
   = note: this error originates in the derive macro `turbo_tasks::NonLocalValue` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `project::Instrumentation: NonLocalValue` is not satisfied
  --> crates/next-api/src/entrypoints.rs:13:26
   |
13 |     pub instrumentation: Option<Instrumentation>,
   |                          ^^^^^^^^^^^^^^^^^^^^^^^ the trait `NonLocalValue` is not implemented for `project::Instrumentation`, which is required by `std::option::Option<project::Instrumentation>: NonLocalValue`
   |
   = help: the following other types implement trait `NonLocalValue`:
             &T
             &mut T
             ()
             (A, Z, Y, X, W, V, U, T)
             (B, A, Z, Y, X, W, V, U, T)
             (C, B, A, Z, Y, X, W, V, U, T)
             (D, C, B, A, Z, Y, X, W, V, U, T)
             (E, D, C, B, A, Z, Y, X, W, V, U, T)
           and 665 others
   = note: required for `std::option::Option<project::Instrumentation>` to implement `NonLocalValue`
note: required by a bound in `entrypoints::_::{closure#0}::DeriveNonLocalValueAssertion::assert_impl_NonLocalValue`
  --> crates/next-api/src/entrypoints.rs:9:1
   |
9  | #[turbo_tasks::value(shared)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `DeriveNonLocalValueAssertion::assert_impl_NonLocalValue`
   = note: this error originates in the derive macro `turbo_tasks::NonLocalValue` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0277`.
error: could not compile `next-api` (lib) due to 3 previous errors
```